pkg/sql/show_create_clauses.go

```diff
@@ -169,16 +169,14 @@ func showForeignKeyConstraint(
 		if err != nil {
 			return err
 		}
-		fkDb, err := lCtx.getDatabaseByID(fkTable.ParentID)
+		fkTableName, err = getTableAsTableName(lCtx, fkTable, dbPrefix)
```

Comment:
does this make more sense to be a method on `TableDescriptor`?

Reply:
I'm not sure, I don't want to move this resolver interface into sqlbase. The functions seem somewhat specific to this process of displaying create statements, so I'm tempted to keep it here.

Reply:
minor nit: can we rename it such that `TableDescriptor` is in the name?

Reply:
done